src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -390,35 +390,12 @@ internal static Delegate GetValueDelegateForFunctionPointer(IntPtr ptr)
 
         private static Guid CreatePIID()
         {
-#if NET
-            if (typeof(T) == typeof(int)) return Nullable_int.IID;
-            if (typeof(T) == typeof(byte)) return Nullable_byte.IID;
-            if (typeof(T) == typeof(bool)) return Nullable_bool.IID;
-            if (typeof(T) == typeof(sbyte)) return Nullable_sbyte.IID;
-            if (typeof(T) == typeof(short)) return Nullable_short.IID;
-            if (typeof(T) == typeof(ushort)) return Nullable_ushort.IID;
-            if (typeof(T) == typeof(char)) return Nullable_char.IID;
-            if (typeof(T) == typeof(uint)) return Nullable_uint.IID;
-            if (typeof(T) == typeof(long)) return Nullable_long.IID;
-            if (typeof(T) == typeof(ulong)) return Nullable_ulong.IID;
-            if (typeof(T) == typeof(float)) return Nullable_float.IID;
-            if (typeof(T) == typeof(double)) return Nullable_double.IID;
-            if (typeof(T) == typeof(Guid)) return Nullable_guid.IID;
-            if (typeof(T) == typeof(global::System.TimeSpan)) return Nullable_TimeSpan.IID;
-            if (typeof(T) == typeof(global::System.DateTimeOffset)) return Nullable_DateTimeOffset.IID;
-            if (typeof(T) == typeof(global::Windows.Foundation.Point)) return IReferenceIIDs.IReferenceOfPoint_IID;
-            if (typeof(T) == typeof(global::Windows.Foundation.Size)) return IReferenceIIDs.IReferenceOfSize_IID;
-            if (typeof(T) == typeof(global::Windows.Foundation.Rect)) return IReferenceIIDs.IReferenceOfRect_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Matrix3x2)) return IReferenceIIDs.IReferenceMatrix3x2_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Matrix4x4)) return IReferenceIIDs.IReferenceMatrix4x4_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Plane)) return IReferenceIIDs.IReferencePlane_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Quaternion)) return IReferenceIIDs.IReferenceQuaternion_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Vector2)) return IReferenceIIDs.IReferenceVector2_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Vector3)) return IReferenceIIDs.IReferenceVector3_IID;
-            if (typeof(T) == typeof(global::System.Numerics.Vector4)) return IReferenceIIDs.IReferenceVector4_IID;
-#endif
-
-            return GuidGenerator.CreateIID(typeof(Nullable<T>));
+            var iid = NullableType.GetIIDForBuiltInType(typeof(T));

```

Comment:
If we make that generic so it can be inlined, the linker might also be able to fully trim out this fallback path for types that are in fact primitive types. Could save some rooted metadata, since I think that `CreateIID` call will root some stuff.

Reply:
Addressed.